values.yaml

```diff
@@ -493,7 +494,8 @@ server:
 
   # Enables a headless service to be used by the Vault Statefulset
   service:
-    enabled: true
+    # @default: global.enabled
+    enabled: "-"
```

Comment:
I'm not sure. On the one hand, it doesn't make sense to have it default `true` if the server would be disabled with `global.enabled = false`. But on the other hand, it feels a little awkward to have `-` have multiple meanings (default to `global.enabled` in previous cases, but here default to `server.enabled` (which then defaults to `global.enabled`)).

I'm open to suggestion, for sure.

Reply:
In my mind, `global.enabled` is a toggle for each of the major components deployed by this chart, specifically affecting `injector.enabled`, `server.enabled`, and `csi.enabled`. Since this is essentially three charts in one, if `global.enabled` directly affects other options inside `server`, that feels like a leaky abstraction in my mind.

And likewise, `injector.enabled`, `server.enabled`, and `csi.enabled` are toggles for everything in their respective components. So to me it seems fine to be to have `server.service.enabled` default to `true`, since that would only be honored if the server component is enabled.

Reply:
Agreed.